homeassistant/components/template/__init__.py

```diff
@@ -101,6 +108,48 @@ async def init_coordinator(hass, conf_section):
         hass.data[DOMAIN] = await asyncio.gather(*coordinator_tasks)
 
 
+def convert_attribute_to_string(attribute):
```

Comment:
Instead of this, modify `RestoreStateData` to use `ExtendedJSONEncoder` which already supports `date`, `datetime` and `timedelta` in a separate PR

Also, is it possible to get `date`, `datetime` and `timedelta` objects as output from templates?

Reply:
I am using ExtendedJSOnEncoder now but had to further extend it as it did not put in type for everything but instead just the date format.